### PR TITLE
feat(statistiques): Add new stats

### PR DESCRIPTION
**Pull request by nkokla**

### CODE REVIEW

File: src/components/Statistics/Statistics.tsx [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
 import { useMemo } from 'react';
 import useSWR from 'swr';
 import Band from './Band';
-import { Container } from './Statistics.style';
+import { Container, GraphsWrapper } from './Statistics.style';
+
+type ReturnApiMatomo = {
+  filters?: { date?: string };
+  nb_uniq_visitors?: string;
+  'Formulaire de test - Envoi'?: { nb_visits?: number | string };
```
Comment (desoindx) [inline src/components/Statistics/Statistics.tsx:13]: Would have been nice to have a type for nb_visits

File: src/pages/api/statistiques/getCountContact.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        'Ville',
+        'Code Postal',
+        'Mode de chauffage',
+        'Type de chauffage',
+        'Distance au réseau',
+        'Gestionnaire',
+        'Status',
+        'en ZDP',
+        'Conso',
+        'Logement',
```
Comment (desoindx) [inline src/pages/api/statistiques/getCountContact.ts:22]: Do you really need to get all this fields ? 
And why don't you use the proper service ?

File: src/pages/api/statistiques/getCountContact.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      ],
+      sort: [{ field: 'Date demandes', direction: 'asc' }],
+    })
+    .all();
+
+  let nbTotal = 0;
+  let nbEligible = 0;
+  let nbUneligible = 0;
+
+  return (records as any[]).reduce((acc, { fields }) => {
```
Comment (desoindx) [inline src/pages/api/statistiques/getCountContact.ts:32]: it's not an any[] but a Demand[]

**Review by desoindx — CHANGES_REQUESTED** [review]

File: src/components/Statistics/Statistics.tsx (hunk 1 above)
Comment (nkokla) [reply src/components/Statistics/Statistics.tsx:13]: Sure 👍

File: src/pages/api/statistiques/getCountContact.ts (hunk 2 above)
Comment (nkokla) [reply src/pages/api/statistiques/getCountContact.ts:22]: No, i don't need all these fields. (It is an WIP for Florence first, to test functional value and i must clean this after his review). 
And I have not use the service because we currently need a email for controle they have an acount (but i will probably update that with anonymous sending -without firstname, lastname and email- when email is not given).

File: src/components/Graph/Graph.style.ts [hunk 4]
```diff
@@ -1,18 +1,14 @@
...
 export const GraphTitle = styled.h3`
   text-align: center;
 `;
 
-export const Container = styled.div`
+export const Container = styled.div<{ large?: boolean }>`
   width: 100%;
+
   @media (min-width: 992px) {
-    width: 50%;
```
Comment (desoindx) [inline src/components/Graph/Graph.style.ts:15]: You don't have any large graph

File: src/components/Statistics/Statistics.tsx [hunk 5]
```diff
@@ -1,44 +1,41 @@
...
-  'Juil',
-  'Aout',
-  'Sept',
-  'Oct',
-  'Nov',
-  'Dec',
-];
+import { Container, GraphsWrapper } from './Statistics.style';
+
+type NbVisitType = { nb_visits?: number | string };
```
Comment (desoindx) [inline src/components/Statistics/Statistics.tsx:10]: try to be consistent, you don't use Type in the other type name. The CamelCase should be enough

Comment (desoindx) [inline src/components/Statistics/Statistics.tsx:10]:
```suggestion
type NbVisit = { nb_visits?: number | string };
```

File: src/components/Statistics/Statistics.tsx [hunk 6]
```diff
@@ -1,44 +1,41 @@
...
 const Statistics = () => {
-  const { data: rawDataActions, error: errorActions } = useSWR(
-    '/api/statistiques/getActions',
-    fetcher
-  );
-  const { data: rawDataVisits, error: errorVisits } = useSWR(
-    '/api/statistiques/getVisitsSummary',
-    fetcher
-  );
+  // Eligibility tests
```
Comment (desoindx) [inline src/components/Statistics/Statistics.tsx:30]: ?

File: src/components/Statistics/Statistics.tsx [hunk 7]
```diff
@@ -47,17 +44,41 @@ const Statistics = () => {
...
+      return [
+        label,
+        entry?.['Formulaire de test - Envoi']?.nb_visits || 0,
+        entry?.['Formulaire de test - Adresse Inéligible']?.nb_visits || 0,
+        entry?.['Formulaire de test - Adresse Éligible']?.nb_visits || 0,
+      ];
+    }),
+  ];
+
+  // Visits
```
Comment (desoindx) [inline src/components/Statistics/Statistics.tsx:73]: useless comment

File: src/core/infrastructure/repository/manager.ts [hunk 8]
```diff
@@ -16,43 +28,52 @@ export const getGestionnaire = (demand: Demand): string | null => {
...
-  const user = await getUser(email);
-
-  if (!user) {
-    return null;
-  }
-
-  const isDemoUser = user.gestionnaire === 'DEMO';
-  const records = await base(tableNameFcuDemands)
-    .select({
+export const getDemands = async (email?: string): Promise<Demand[] | null> => {
```
Comment (desoindx) [inline src/core/infrastructure/repository/manager.ts:31]: Why did you change all of this ?

File: src/pages/api/demands/index.ts [hunk 9]
```diff
@@ -85,15 +85,15 @@ export default async function demands(
...
   try {
     const session = await getSession({ req });
-    const email = session?.user?.email;
-    if (!email) {
-      return res.status(204).json([]);
-    }
-
+    const email = session?.user?.email ?? undefined;
     if (req.method === 'GET') {
       return get(res, email);
```
Comment (desoindx) [inline src/pages/api/demands/index.ts:90]: That mean that anyone without account can get our demands...
I'm not okay with this !

File: src/pages/api/statistiques/getCountContact.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
+import { getDemands } from '@core/infrastructure/repository/manager';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+const get = async (res: NextApiResponse) => {
+  let nbTotal = 0;
+  let nbEligible = 0;
+  let nbUneligible = 0;
+
+  const demands = (await getDemands())?.reverse()?.reduce((acc, fields) => {
```
Comment (desoindx) [inline src/pages/api/statistiques/getCountContact.ts:9]: why those '?'

File: src/pages/api/statistiques/getMonthContact.ts [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    return !date
+      ? acc
+      : {
+          ...acc,
+          [key]: value,
+        };
+  }, {});
+  return res.status(200).json(demands);
```
Comment (desoindx) [inline src/pages/api/statistiques/getMonthContact.ts:36]: thats basically the same code as CountContact... you can merge those two

**Review by desoindx — CHANGES_REQUESTED** [review]

File: src/pages/api/demands/index.ts (hunk 9 above)
Comment (nkokla) [reply src/pages/api/demands/index.ts:90]: Everyone can see the demands on the map.

File: src/core/infrastructure/repository/manager.ts (hunk 8 above)
Comment (nkokla) [reply src/core/infrastructure/repository/manager.ts:31]: All these changes are for call without email. In this case, the result is anonymized. (The personal data are not send)

File: src/pages/api/demands/index.ts (hunk 9 above)
Comment (desoindx) [reply src/pages/api/demands/index.ts:90]: they can see the fields we want them to see... they don't see the whole demand

File: src/core/infrastructure/repository/manager.ts (hunk 8 above)
Comment (desoindx) [reply src/core/infrastructure/repository/manager.ts:31]: you don't need to complexify the code then... juste to consider people without email as demoUser (and change the name)

Comment (nkokla) [reply src/core/infrastructure/repository/manager.ts:31]: Sure ? 
This is not exactly the same. 
For demo account, we send some fake value for personnal data.
For free account, we don't send personnal data.
